### PR TITLE
CI - phase #1

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
Upon PR, build XPMEM on multiple kernel versions in Docker.

### CODE REVIEW

**Comment by Alexey-Rivkin** [comment]
@yosefe, @artemry-nv please review

File: ci/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+              kernel_ver=$(rpm -qa | grep kernel-devel | cut -d'-' -f3-)
+              ./configure --with-kerneldir=/usr/src/kernels/${kernel_ver}
+              make
+            displayName: Build CentOS
+            condition: contains(variables['build_container'], 'centos')
+
+          - bash: |
+              set -eEx
+              ./autogen.sh
+              kernel_ver=$(dpkg -l | grep 'linux-headers-.*-generic' | awk '{print $2}')
```
Comment (brminich) [inline ci/azure-pipelines.yml:67]: do we really need `generic` here? For instance on BF it is:
```
ii  linux-headers-5.4.0-1050-bluefield 5.4.0-1050.56                                            arm64        Linux kernel headers for version 5.4.0 on ARMv8 SMP
ii  linux-headers-bluefield            5.4.0.1050.48                                            arm64        Linux kernel headers for Mellanox Bluefield systems.
```

Comment (Alexey-Rivkin) [reply ci/azure-pipelines.yml:67]: To be run in Docker, so the output doesn't match what's installed on hosts.

Comment (brminich) [reply ci/azure-pipelines.yml:67]: ok

**Comment by brminich** [comment]
@tvegas1, @artemry-nv please review as well

**Comment by artemry-nv** [comment]
@B-a-S please review also

**Comment by yosefe** [comment]
Merging this to have initial CI. Pls apply review fixes in another PR.

**Comment by tvegas1** [comment]
Looks good

**Pull request merged**